larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,287 @@
+        raise NotImplementedError("ExcelReport class cannot be instantiated because xlwings is not installed")
+
+
+if xw is not None:
+    from xlwings.constants import LegendPosition, HAlign, VAlign, ChartType, RowCol, Direction
+    from larray.inout.xw_excel import open_excel
+
+    def create_instance(app_id):
+        import pythoncom
+        from win32com.client import gencache
+        # EnsureDispatch use early binding but will reuse an existing instance,
+        # while DispatchEx creates an new application but uses late binding.
+        # We want both a new instance and use early binding.
+        app_id = pythoncom.CoCreateInstanceEx(app_id, None, pythoncom.CLSCTX_SERVER, None,
+                                              (pythoncom.IID_IDispatch,))[0]
+        if gencache.is_readonly:
+            # needed for "frozen" apps
+            gencache.is_readonly = False
+            gencache.Rebuild()
+        return gencache.EnsureDispatch(app_id)
+
+    # open an excel instance via win32com "EnsureDispatch" so that we flag Excel as using early binding.
+    # it is much faster (~85% faster in this case -- 2.39s -> 1.28s) than late binding BUT sometimes (I don't know the
+    # precise circumstances yet -- when Excel is updated?) it gets corrupt and nothing works until you re-generate them.
+    # Note that this only needs to be done _once_ per machine.
+
+    # This can also be done explicitly on the command line in two different ways:
+    # * to generate one big file with bindings for all Excel objects:
+    #   python -m win32com.client.makepy Excel.Application
+    # * to flag "Excel" as using early binding. Bindings will be generated as many small files as needed:
+    #   python -c "from win32com.client import gencache;gencache.EnsureDispatch('Excel.Application')"
+    # In case something goes wrong, one can go back to late binding by removing generated bindings from:
+    #   'C:\\Users\\<user>\\AppData\\Local\\Temp\\gen_py\\3.6'
+    try:
+        app = create_instance('Excel.Application')
+    except AttributeError:
+        pathdir = os.path.join(os.path.expanduser('~'), 'AppData', 'Local', 'Temp', 'gen_py')
+        rmtree(pathdir)
+        app = create_instance('Excel.Application')
+
+    class AbstractExcelItem(object):
+        def __init__(self, top, left, width, height):
+            self.top = top
+            self.left = left
+            self.width = width
+            self.height = height
+
+        def dump(self, sheet, data_sheet, row):
+            raise NotImplementedError()
+
+    _default_items_size = {}
+
+    class TitleItem(AbstractExcelItem):
+        def __init__(self, text, fontsize, top, left, width, height):
+            AbstractExcelItem.__init__(self, top, left, width, height)
+            self.text = str(text)
+            self.fontsize = fontsize
+
+        def dump(self, sheet, data_sheet, row):
+            data_range = data_sheet.range
```

Comment:
I still don't get why you didn't use the raw win32com code I produced, which was faster. The nan problem only applies to the "data.dump()" line in GraphItem.dump(), right?